Mathlib/Algebra/Group/IsTorsionFree/Defs.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2024 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Algebra.Group.Defs
+import Mathlib.Tactic.MkIffOfInductiveProp
+
+/-!
+# Definitions about torsion free monoids
+
+In this file we define typeclasses for torsion free additive and multiplicative monoids.
+-/
+
+/-- An additive monoid is called *torsion free*,
+if `n • x = 0`, `n : ℕ`, is possibly only if `n = 0` or `x = 0`. -/
+class IsAddTorsionFree (M : Type*) [AddMonoid M] : Prop where
+  eq_zero_of_nsmul_eq_zero (x : M) (n : ℕ) : n • x = 0 → n ≠ 0 → x = 0
```

Comment:
Is this the same as `NoZeroSMulDivisors`?

Reply:
Indeed, it seems like we could just multiplicativize `NoZeroSMulDivisors` and then make `IsMulTorsionFree` an `abbrev class`.